### DISCUSSION TITLE
ask help with esp32 s2 mini + sx1278 (spi trouble?)

### DISCUSSION THREAD

**Discussion by whoim2** [jgromes/RadioLib#809] [Q&A]
[This board (pinout)](https://www.studiopieters.nl/wp-content/uploads/2022/08/WEMOS-ESP32-S2-MINI.png)

module: ebyte e32-400m30s
connected:
`/* MISO 37
 * MOSI 35
 * SCK  36
 * CS   34
 * DIO0 21
 * RST  18
 * DIO1 17
 * RXEN 40
 * TXEN 39
 */`

try to init
```
SPIClass *spi2 = new SPIClass(FSPI);
SPISettings spiSettings(2000000, MSBFIRST, SPI_MODE0);
SX1278 radio = new Module(34, 21, 18, 17, spi2, spiSettings);
..
void setup() {
spi2.begin(36, 35, 37, 34);
int state = radio.beginFSK();
}
```

state = -2, debug:
`SX127x not found! (10 of 10 tries) RADIOLIB_SX127X_REG_VERSION == 0x0000, expected 0x0012
No SX127x found!`

Its trouble with spi init?

**Comment by whoim2** [comment]
im swap mosi&miso pins at spi.begin ;)

**Reply by jgromes** [reply]
Glad you were able to resolve the issue!
